### ISSUE TITLE
Configuring server.js

### ISSUE THREAD

**Issue by alineborak** [bahmutov/nock-getServerSideProps#1]
Hello Gleb,
I'm testing this configuration on the project that I work for, however I'm not sure how to adjust my current project server.js file adding the suggested config you have here.
On server.js file, line 11, when it comes to preparing the app:
- Should I add a new `app.prepare().then(() => {` ... ?

**Comment by bahmutov** [comment]
well, I cannot say what is going on in your code or your application, I would think yes? I mean that's what I have working...

**Comment by alineborak** [comment]
This is what I currently have in the project, I added some lines as you had shared (checking to not repeat with what already was there):

```
const express = require('express');
const path = require('path');
const next = require('next');
const { createProxyMiddleware } = require("http-proxy-middleware");
const requestIp = require('request-ip');
const getConfig = require('./next.config');
const { createServer } = require('http');
const { parse } = require('url');

const port = process.env.PORT || 3000
const dev = process.env.NODE_ENV !== 'production'
const app = next({ dev, dir: './src' })
const handle = app.getRequestHandler()

const envConfig = {
    CLOUDFRONT_URL: getConfig ? getConfig.publicRuntimeConfig.CLOUDFRONT_URL : process.env.CLOUDFRONT_URL
}

const proxyResHandle = (proxyRes, req, res) => {
    if (proxyRes.req.path.includes('gz')) {
        proxyRes.headers['Content-Disposition'] = 'attachment';
        proxyRes.headers['Content-Type'] = 'application/gzip';
    }
}

app.prepare().then(() => {
    const server = express();
    server.disable("x-powered-by");

    server.use(express.static(path.join(__dirname, "/src/public")));
    server.use(['/extern/**', '/sitemap/**', '/sitemap.xml'], createProxyMiddleware({
        target: `${envConfig.CLOUDFRONT_URL}`,
        changeOrigin: true,
        secure: true,
        pathRewrite: {
            '^/sitemap.xml': '/sitemap/sitemap.xml'
        },
        onProxyRes: proxyResHandle
    }));

    app.prepare().then(() => {
        createServer((req, res) => {
            // Be sure to pass `true` as the second argument to `url.parse`.
            // This tells it to parse the query portion of the URL.
            const parsedUrl = parse(req.url, true)
            handle(req, res, parsedUrl)
        }).listen(3000, (err) => {
            if (err) throw err
            console.log('> Ready on http://localhost:3000')
        })
    })

    server.all('*', async (req, res) => {
        return handle(req, res);
    });

    server.listen(port, '0.0.0.0', (err) => {
        if (err) throw err
        console.log(`> Ready on http://localhost:${port}`)
    })
}).catch(err => {
    console.log('Error:::::', err)
})
```

**Comment by bahmutov** [comment]
So, is it not working? What’s the actual issue?

**Comment by alineborak** [comment]
When I run the test, i get this error. Not sure if it's related but...
![Screenshot 2021-10-20 at 15 41 18](https://user-images.githubusercontent.com/36476341/138104361-318180ed-1699-4057-8ab9-2b5f8ac47eb2.png)
t

**Comment by bahmutov** [comment]
So open the devtools look at the exception and where it happens and the data it is trying to destruct

**Comment by alineborak** [comment]
I tried to do that, but after that my localhost could not open anymore, something with the api connection:
<img width="991" alt="Screenshot 2021-10-21 at 09 15 17" src="https://user-images.githubusercontent.com/36476341/138247794-816a7748-42c0-4c8a-9f7a-0f9951e5c2b5.png">

After I removed the config I had added, that solved it :( 
But then I'm back to square zero.
I'm quite lost on this step by step. I tried to follow the blog documentation, and after I tried to follow the video, but some things are slightly confusing for someone doing that for the very first time.
